docs/design/rule-manager/README.md

```diff
@@ -0,0 +1,75 @@
+# Rule manager
+
+## Overview
+
+Rule manager is a separate open-kilda component responsible for building rules. The main goal of this component is to encapsulate all rule related logic into one place. All other components (flow CRUD, flow/switch validation and sync, default rules management, etc) should use rule manager API to build rules.
+
+## Details
+
+Rule manager implemented as a separate library module. It should support input data from database (for internal system calls) and data provided by user in json format (for manual calls). Manual calls should have CLI options to support data input from a file. Rule manager should use adapters to encapsulate input data and provide the same results for different input types. 
+
+The main idea is to build rules as close as possible to the actual OpenFlow representation so Floodlight may just translate messages directly into switch commands. Rule manager doesn't send any commands to anywhere it's responsible only for building rules. Sending results to floodlight is out of scope and should be covered by calling component.
+
+![Sequence diagram](sequence-diagram.png "sequence diagram")
+
+## API
+
+Rule manager should be able to process several types of requests:
+* Build rules for flow.
+
+  Required parameters: flow with paths, switch properties for used switches.
```

Comment:
We should use path (with flow) or path-pair (with flow), but not flow with paths.

Reply:
I guess it's the same data. Rewrited.